client/src/elm/Pages/ChildScoreboard/Activity/Model.elm

```diff
@@ -2,29 +2,91 @@ module Pages.ChildScoreboard.Activity.Model exposing (..)
 
 import Backend.Entities exposing (..)
 import Backend.Measurement.Model exposing (..)
+import DateSelector.Model exposing (DateSelectorConfig)
 import EverySet exposing (EverySet)
 import Gizra.NominalDate exposing (NominalDate)
-import Measurement.Model exposing (NCDAData, NCDAForm, NCDAStep, emptyNCDAData)
+import Measurement.Model
+    exposing
+        ( ImmunisationTask(..)
+        , NCDAData
+        , NCDAForm
+        , NCDAStep
+        , VaccinationForm
+        , VaccinationFormViewMode
+        , VaccinationProgressDict
+        , emptyNCDAData
+        , emptyVaccinationForm
+        )
 import Pages.Page exposing (Page)
 
 
 type alias Model =
     { ncdaData : NCDAData
+    , immunisationData : ImmunisationData
     }
 
 
 emptyModel : Model
 emptyModel =
     { ncdaData = emptyNCDAData
+    , immunisationData = emptyImmunisationData
+    }
+
+
+type alias ImmunisationData =
+    { bcgForm : ChildScoreboardVaccinationForm
+    , dtpForm : ChildScoreboardVaccinationForm
+    , ipvForm : ChildScoreboardVaccinationForm
+    , mrForm : ChildScoreboardVaccinationForm
+    , opvForm : ChildScoreboardVaccinationForm
+    , pcv13Form : ChildScoreboardVaccinationForm
+    , rotarixForm : ChildScoreboardVaccinationForm
+    , activeTask : Maybe ImmunisationTask
+    }
+
+
+type alias ChildScoreboardVaccinationForm =
+    VaccinationForm Msg
+
+
+emptyImmunisationData : ImmunisationData
+emptyImmunisationData =
+    { bcgForm = emptyVaccinationForm
+    , dtpForm = emptyVaccinationForm
+    , ipvForm = emptyVaccinationForm
+    , mrForm = emptyVaccinationForm
+    , opvForm = emptyVaccinationForm
+    , pcv13Form = emptyVaccinationForm
+    , rotarixForm = emptyVaccinationForm
+    , activeTask = Nothing
     }
 
 
 type Msg
-    = SetActivePage Page
+    = NoOp
```

Comment:
Where do we use this `NoOp`? It seems to be an outlier compared to the other messages. Do we really need it?

Reply:
Ah, I see, is it temporary then?

Reply:
@AronNovak 

Well, no, it's not temporary. This is 'empty' message - similar to concept of `emptyNode`.
There are rare cases where we need the `NoOp` (no operation), as some of logical pathes should result in action while others are not. For example:

![Selection_2100](https://github.com/TIP-Global-Health/eheza-app/assets/17882245/b349694b-8d68-495c-9906-381d1d51b1d3)